src/platform_impl/ios/view.rs

```diff
@@ -235,6 +237,7 @@ unsafe fn get_window_class() -> &'static Class {
                             device_id: RootDeviceId(DeviceId { uiscreen }),
                             id: touch_id,
                             location: (location.x as f64, location.y as f64).into(),
+                            force: (force / max_force) as _,
```

Comment:
This is probably lossy and inconsistent across devices. From the [Apple docs](https://developer.apple.com/documentation/uikit/uitouch/1618110-force?language=objc).

> The force of the touch, where a value of 1.0 represents the force of an average touch (predetermined by the system, not user-specific).

If apple provides an API for `maximumPossibleForce` then it's likely to be different across devices, meanwhile `1.0` is always the force of an average touch. This means users will have to reverse the above logic to figure out if the touch is a force touch or not.

A couple of other questions:
- Do other platforms do things similarly to iOS (1.0 is average, everything is a float)?
- What scale is iOS using (linear/logarithmic/etc)?

Reply:
I have similar concerns here. I didn't look at other platforms at length, but on Windows it appears that the force is reported as an integer between 0 and 1024 where 1024 is the maximum pressure applicable. I don't know what scale iOS is using for the force here. I'm not sure if it's possible to make this behave identically across platforms. What do you suggest?

Reply:
On iOS it [https://forums.developer.apple.com/thread/31775](looks like) values are calibrated to be consistent across devices, however different devices may have different max force.

Android appears to have the 0-1 clamping behavior, but touch pressure scaling can cause > 1 values to be returned.

> I'm not sure if it's possible to make this behave identically across platforms.

Agreed. That's too bad :/.

> What do you suggest?

I suggest we just pass the float along with no scaling for now, and add some documentation explaining the values it might have on each platform.

Reply:
> I suggest we just pass the float along with no scaling for now, and add some documentation explaining the values it might have on each platform.

I strongly disagree with that suggestion. If we're exposing an API that's common across platforms, the semantics of that API should differ as little as possible between platforms - doing otherwise creates an API that's scary to use and will likely cause compatibility bugs downstream.

I'm not opposed to exposing a raw pressure value that is platform dependent (some sort of `TouchExt` trait would probably be best for this), but platform-dependent behaviors should be opt-in when possible, not the default.

Reply:
We should then also add a function returning the maximum possible force on iOS.

Reply:
If we don't normalize the value, the user will need to normalize the returned value in their application using platform-specific extension traits if a value in the 0..=1 range is required. This happens to be my use case.

Apple cautions against normalizing the force in the [forum thread](https://forums.developer.apple.com/thread/31775) linked above. They say this is because they configure this range by setting `maximumPossibleForce` so that, say, a value of 3.0 will roughly match the same force across all their devices. I don't think this is an issue for us, as we don't expect a value of 0.5 to match the same amount of force being applied to the device across different devices anyway, say, iPhones and Microsoft Surface tablets.

I think I'm leaning towards normalizing the force to the 0..=1 range, and providing platform-specific extensions for iOS to access the raw values for the purpose of detecting the force touch activation point.

Reply:
The `force: f64` value is difficult to use correctly without more information regardless of what we do. `winit` users should know something about the os, os settings, or input device.

IMO constraining the value from `0-1` would make folks assume that the value is useful without platform specific code, and gains little. iOS is the only platform that uses touch force as a core part of its UX design and making `force` easy to use _incorrectly_ on that platform by constraining the range feels wrong to me. Are you sure your usage of `force` won't cause someone with a different device to have to mash the screen really hard? Max possible force also can differ from touch to touch. [Apple pencil](https://developer.apple.com/documentation/uikit/pencil_interactions/handling_input_from_apple_pencil/computing_the_perpendicular_force_of_apple_pencil?language=objc) requires more care to get a correct force value.

Maybe force could be more than just a float, so that it's obvious that some nuanced interpretation of the value is required.

Names chosen arbitrarily:
```rust
enum Force {
  IOS { force: f64, max_possible: f64, altitude: Option<f64> },
  Normalized { force: f64 },
  ...
}
impl Force {
  fn normalized(&self) -> f64;
}
```

Reply:
That makes a lot of sense to me. I've added a commit that adds a variant for the calibrated force returned on iOS. I think the ergonomics of this are pretty good, and I've added some documentation to make this easier to understand for users. What do you think, @Osspial?

Reply:
I'm down with that design.

Reply:
I've rebased this against `master`. Should be good to merge, unless something else comes up?